micromanager_gui/main_window.py

```diff
@@ -92,8 +94,10 @@ def __init__(
         sig.systemConfigurationLoaded.connect(self._on_system_cfg_loaded)
         sig.XYStagePositionChanged.connect(self._on_xy_stage_position_changed)
         sig.stagePositionChanged.connect(self._on_stage_position_changed)
+        sig.channelGroupChanged.connect(self._refresh_channel_list)
         sig.exposureChanged.connect(self._on_exp_change)
         sig.frameReady.connect(self._on_mda_frame)
+        sig.propertyChanged.connect(self._on_offset_status_changed)
```

Comment:
this connection is too broad.  This will perform the logic in `_on_offset_status_changed` whenever literally _any_ property in mmcore changes... which will happen all the time.  Create an intermediate callback that checks the name of the property being changed, and does nothing if it doesn't need to:

```python
    sig.propertyChanged.connect(self._on_prop_changed)

def _on_prop_changed(self, dev_name: str, prop_name: str, new_value: str):
    if devname_or_propname_indicates_perfect_focus:
        self._on_offset_status_changed()
```